src/components/Common/AvatarEditModal.tsx

```diff
@@ -76,13 +85,36 @@ const AvatarEditModal = ({
   );
   const { t } = useTranslation();
   const [isDragging, setIsDragging] = useState(false);
+  const { preferredDeviceId, setDeviceId } = usePreferredMediaDevice();
+
+  useEffect(() => {
+    const getDevices = async () => {
+      try {
+        const mediaDevices = await navigator.mediaDevices.enumerateDevices();
+        const videoDevices = mediaDevices.filter(
+          ({ kind }) => kind === "videoinput",
+        );
+        setDevices(videoDevices);
+        if (videoDevices.length > 0) {
+          const initialDeviceId =
+            preferredDeviceId != null
```

Comment:
Let's also add a check to ensure the device is present in videoDevices list (in case, user disconnects their preferred camera and it's no longer accessible).

Reply:
Take a following situation:
(1) User already selected a preferred device (and now stored in local storage)
(2) User removes the device (local storage still has the device Id)
(3) Now you are using that device Id which would not be present in the the devices list.